bookkeeper-server/src/test/java/org/apache/bookkeeper/test/BookieClientTest.java

```diff
@@ -228,22 +224,19 @@ public void testWriteGaps() throws Exception {
         }
     }
 
-    private ChannelBuffer createByteBuffer(int i, long lid, long eid) {
-        ByteBuffer bb;
-        bb = ByteBuffer.allocate(4 + 24);
-        bb.putLong(lid);
-        bb.putLong(eid);
-        bb.putLong(eid-1);
-        bb.putInt(i);
-        bb.flip();
-        return ChannelBuffers.wrappedBuffer(bb);
+    private ByteBuf createByteBuffer(int i, long lid, long eid) {
+        ByteBuf bb = Unpooled.buffer(4 + 16);
+        bb.writeLong(lid);
+        bb.writeLong(eid);
```

Comment:
@kishorekasi this is the cause of the failure of test testWriteGaps() (I remember the same problem with that test was solved by @sijie some time ago)
The fix is very simple, just update the method with the current master code adapted to ByteBuf
```
        ByteBuf bb = Unpooled.buffer(4 + 24);
        bb.writeLong(lid);
        bb.writeLong(eid);
        bb.writeLong(eid-1);
        bb.writeInt(i);
        return bb;
```
I hope that helps